Automattic/wp-calypso
# Plans 2023: data store slugs and more

Related to #76024 

## Proposed Changes

Continuation of #76024 to introduce the plans data store into the 2023 plan grids. 
- Introduces more slugs to cover the rest of the plans
- `billingTerm` included in the plan product structure (used in follow up PR)
- a selector to fetch plan product given a respective slug (used in follow up PR)

### Author Notes

The draft P2 post ( dvytD-p2  ) contains details about this work and motivation. For the time being, we move enough content into the plans data store to surface it for usage in the plans 2023 grids (continuing on how it's already set up to work).

## Testing Instructions



There are no UI integrations. One way to test:
- go to `/plans/[SITE]`
- open Redux dev tools and inspect the latest image for `automattic/onboard/plans`
- under `State`, confirm all the variant `planProducts` for different periods are visible
- confirm `billingTerm` is included in that structure

<img width="875" alt="Screenshot 2023-04-26 at 12 09 42 PM" src="https://user-images.githubusercontent.com/1705499/234528693-0edc3dc4-14c7-4b89-9c65-eff6dc6e6c07.png">



## Pre-merge Checklist
